RevenueCatUI/Templates/V2/Components/Image/ImageComponentView.swift

```diff
@@ -54,7 +54,19 @@ struct ImageComponentView: View {
             )
         ) { style in
             if style.visible {
-                if let maxWidth = self.maxWidth {
+                let expectedSize = CGSize(
+                    width: self.imageSize(style: style).width,
+                    height: self.imageSize(style: style).height
+                )
+
+                ZStack {
+                    GeometryReader { proxy in
+                        Color.clear
+                            .onAppear {
+                                self.maxWidth = proxy.size.width
+                            }
+                    }
```

Comment:
Since we need to do this pretty often, I have a helper coming soon. Though I need to make it a modifier. May be good to use it in this PR too and we can just clean up the merge conflicts in my PR later:


https://github.com/RevenueCat/purchases-ios/blob/9e871323882527c55f491bc26851efb15039b31d/RevenueCatUI/Templates/V2/ViewHelpers/SizeReader.swift#L18-L52